utils/python/CIME/XML/env_mach_specific.py

```diff
@@ -116,7 +116,8 @@ def _compute_actions(self, nodes, child_tag):
             if (self._match_attribs(node.attrib)):
                 for child in node:
                     expect(child.tag == child_tag, "Expected %s element" % child_tag)
-                    result.append( (child.get("name"), child.text) )
+                    if (self._match_attribs(child.attrib)):
```

Comment:
Bug fix: need to check child attributes. I left in the code that checks the parent's attributes, perhaps that can be removed.